### ISSUE TITLE
Amusing and wrong error message

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rust#51874]
```rust
fn main() {
    let a = (1.0).pow(1.0);
}
```

```
error[E0689]: can't call method `pow` on ambiguous numeric type `{float}`
 --> src/main.rs:2:19
  |
2 |     let a = (1.0).pow(1.0);
  |                   ^^^
help: you must specify a concrete type for this numeric value, like `f32`
  |
2 |     let a = (1.0)_f32.pow(1.0);
  |             ^^^^^^^^^
```

Please note the `(1.0)_f32`

**Comment by estebank** [comment]
The literal's span should point inside the parenthesis. Probably needs a change to `libsyntax/parser/parse.rs`.

**Comment by euclio** [comment]
I'm interested in working on this.

**Comment by estebank** [comment]
@euclio feel free to reach out for any help you might need!

I believe that you should verify if changing the span being used in the AST for literals changes the suggestion output. If it doesn't, it means that the problem is not at the parser level, but when lowering to HIR. If that is the case, we need to dig deeper, but I'll be here to help you with that.

https://forge.rust-lang.org/x-py.html
